### ISSUE TITLE
Rust format not working, rustfmt internal error[internal]: left behind trailing whitespace

### ISSUE THREAD

**Issue by yuribudilov** [rust-analyzer/rust-analyzer#11143]
Hello everyone,

This may not be a bug but a behavioral issue or may be I am doing something wrong.

The rust format does not work when I save on some of my folders (not all of them, so it's not a settings.json issue on my part)

So I tried using the command line rustfmt .\src\main.rs instead.
See below. I received the following error: note the **^^** below with **2 trailing spaces at the end of the source code line**.
```

rustfmt --color always .\src\main.rs

error[internal]: left behind trailing whitespace

>   --> \\?\C:\Yuri\rust\photos_r_similar\src\main.rs:61:61:19
>      |
> 61 |     let image2   =  
>      |                            ^^
>      |
> warning: rustfmt has failed to format. See previous 1 errors.
> 
> 
```

So I manually corrected the Rust code in VSCode editor i.e. I **removed the 2 trailing spaces after the = sign** above.

After that, Rust code formatting works in VS Code !! 
It behaves as if it calls rustfmt from VSCode and it gets rustfmt internal error trying to format and this is why it does not format some of my Rust code but not all of them (since most of them do not cause internal errors from rustfmt?).

Any ideas what is happening?
Should I open an issue with **rustfmt** (I dont know where this is done, on **git** somewhere??)

thank you very much

**Comment by lnicola** [comment]
> It behaves as if it calls rustfmt from VSCode and it gets rustfmt internal error trying to format and this is why it does not format some of my Rust code

That is exactly what's happening. There's no formatting support in rust-analyzer (and if there was, it would have to produce identical results to `rustfmt` or people would complain).

The rustfmt repo is https://github.com/rust-lang/rustfmt. But it might be already reported. One thing to keep in mind is that it only works on syntactically valid code.

**Comment by yuribudilov** [comment]
wonderful response, thank you. Happy 2022.
